synapse/storage/databases/main/lock.py

```diff
@@ -96,6 +95,10 @@ def __init__(
 
         self._acquiring_locks: Set[Tuple[str, str]] = set()
 
+        self._clock.looping_call(
+            self._reap_stale_read_write_locks, _LOCK_TIMEOUT_MS / 10.0
+        )
```

Comment:
Is every worker going to be doing this reaping? I guess that's fine since the reaps are idempotent.

Reply:
Yeah, good point but I don't think it matters in practice.